jekyll/_cci2/use-the-circleci-cli-to-split-tests.adoc

```diff
@@ -0,0 +1,189 @@
+- `**` matches any sequence of characters (including path separators)
+- `?` matches any single character (excluding path separators)
+- `[abc]` matches any character (excluding path separators) against characters in brackets
+- `{foo,bar,...}` matches a sequence of characters, if any of the alternatives in braces matches
+
+To check the results of pattern-matching, use the `echo` command.
+
+```yaml
+# ~/.circleci/config.yml
+version: 2
+jobs:
+  test:
+    docker:
+      - image: cimg/<language>:<version TAG>
+        auth:
+          username: mydockerhub-user
+          password: $DOCKERHUB_PASSWORD  # context / project UI env-var reference
+    parallelism: 4
+    steps:
+      - run:
+          command: |
+            echo $(circleci tests glob "foo/**/*" "bar/**/*")
+            circleci tests glob "foo/**/*" "bar/**/*" | xargs -n 1 echo
+```
+
+[#split-tests]
+== 2. Split tests
+
+To split tests, pass in a list of tests to the `circleci tests split` command.
+
+The following test splitting options are available:
+
+* Alphabetically by name (by default, if none specified)
+* `--split-by=timings` (recommended)
+* `--split-by=filesize`
+
+[#split-by-name]
+=== a. Split by name (default)
+
+By default, if you do not specify a method using the `--split-by` flag, `circleci tests split` expects a list of filenames or classnames and splits tests alphabetically by test name. There are a few ways to provide this list:
+
+* Pipe a glob of test files, as demonstrated in the above section.
+```shell
+circleci tests glob "test/**/*.java" | circleci tests split
+```
+
+* Create a text file with test filenames.
+```shell
+circleci tests split test_filenames.txt
+```
+
+* Provide a path to the test files.
+```shell
+circleci tests split < /path/to/items/to/split
+```
+
+[#split-by-timing-data]
+=== b. Split by timing data
+
+The best way to optimize your test suite across a set of parallel executors is to split your tests using timing data. This will ensure the tests are split in the most even way, leading to a shorter test time.
+
+To split by test timings, use the `--split-by` flag with the `timings` split type. 
+
+```shell
+circleci tests glob "**/*.go" | circleci tests split --split-by=timings
+```
+
+WARNING: If you do not use `store_test_results`, there will be no timing data available to split your tests.
+
+On each successful run of a test suite, CircleCI saves timings data from the directory specified by the path in the xref:configuration-reference#storetestresults[`store_test_results` step]. This timings data consists of how long each test took to complete per filename or classname.
+
+The available timings data will then be analyzed and your tests will be split across your parallel-running containers as evenly as possible.
+
+NOTE: If no timing data is found, you will receive a message: `Error autodetecting timing type, falling back to weighting by name.`. The tests will then be split alphabetically by test name.
```

Comment:
I'm not sure the specifics around the context, but it seems we use both "timings" and "timing" throughout the pages for this. Is the context specific on when it should be pluralized or not? I'm not sure why sometimes it's pluralized and sometimes it's not. The pluralization version just sounds off to me. For example the section is called "Split by timing data" but the content says "timings data" in a few places.

Reply:
@devinmarieb I just checked a few blog posts and looks like we use "timing data" -- I'll standardize over here too